rx888_source/src/main.cpp

```diff
@@ -76,188 +159,261 @@ class RX888SourceModule : public ModuleManager::Instance {
-                    _this->realStream.swap(rLen / 2);
+            sddc_set_tuner_rf_attenuation(_this->sddc, _this->atten_rf);
+            sddc_set_tuner_if_attenuation(_this->sddc, _this->atten_if);
+        }
+        ImGui::Text("BiasT HF");
+        ImGui::SameLine();
+        if (ImGui::Checkbox(CONCAT("##_rx888_bias_hf_", _this->name), &_this->hf_bias)) {
+            sddc_set_hf_bias(_this->sddc, _this->hf_bias ? 1 : 0);
+        }
+        ImGui::Text("BiasT VHF");
+        ImGui::SameLine();
+        if (ImGui::Checkbox(CONCAT("##_rx888_bias_vhf_", _this->name), &_this->vhf_bias)) {
+            sddc_set_vhf_bias(_this->sddc, _this->vhf_bias ? 1 : 0);
+        }
+        if(sddc_support_128adc(_this->sddc) > 0){
+            ImGui::Text("ADC 128M");
+            ImGui::SameLine();
+            if (ImGui::Checkbox(CONCAT("##_rx888_adc_128_", _this->name), &_this->adc_overclock)) {
+#ifdef LIBSDDC_RAW
+                sddc_set_adc_rate(_this->sddc, _this->adc_overclock ? 2*64000000 : 64000000);
+                _this->sampleRate = _this->adc_overclock ? 64000000 : 32000000;
+                core::setInputSampleRate(_this->sampleRate >> _this->deciminator);
+#else
+                if(_this->running)
+                    sddc_stop_streaming(_this->sddc);
+                sddc_set_sample_rate(_this->sddc, _this->sampleRate*2);
+                core::setInputSampleRate(_this->sampleRate);
+                if(_this->running)
+                    sddc_start_streaming(_this->sddc);
+#endif
+                _this->sampleRateListTxt = "";
+                for (int i = 0; i < (_this->adc_overclock ? 6 : 5); i++) {
+                    _this->sampleRateListTxt += sampleRatesTxt[i];
+                    _this->sampleRateListTxt += '\0';
                 }
             }
-
-            // Start next transfer
-            context = EndPt->BeginDataXfer((PUCHAR)buffer, realBlockSize * 2, &inOvLap);
         }
-
-        delete[] buffer;
-
-        // Stop FX3 chip
-        fx3Control(STOPFX3);
+        ImGui::Text("ADC Dither");
+        ImGui::SameLine();
+        if (ImGui::Checkbox(CONCAT("##_rx888_adc_dither_", _this->name), &_this->adc_dither)) {
+            sddc_set_adc_dither(_this->sddc, _this->adc_dither ? 1 : 0);
+        }
+        ImGui::Text("ADC Random");
+        ImGui::SameLine();
+        if (ImGui::Checkbox(CONCAT("##_rx888_adc_random_", _this->name), &_this->adc_random)) {
+            sddc_set_adc_random(_this->sddc, _this->adc_random ? 1 : 0);
+        }
     }
 
-    static void _worker(RX888SourceModule* _this) {
+#ifdef LIBSDDC_RAW
+    static void asyncHandler(uint32_t len, uint8_t *buf, void *ctx) {
+        RX888SourceModule* _this = (RX888SourceModule*)ctx;
         dsp::complex_t* iqbuffer = new dsp::complex_t[STREAM_BUFFER_SIZE];
+        int blockSize = len/sizeof(float);
+        for (int i = 0; i < blockSize; i++) {
+            iqbuffer[i].re = (float)buf[i*sizeof(float)] / 32768.0f;
```

Comment:
Use volk for type conversion instead of loop

Reply:
Fixed